geomeppy/view_geometry.py

```diff
@@ -121,11 +147,34 @@ def _get_collections(idf, opacity=1):
     return walls, roofs, floors, windows, shading
 
 
-def _get_collection(surface_type, surfaces, opacity, facecolor, edgecolors="black"):
+def _get_collection(surface_type, surfaces, opacity, zones, facecolor, edgecolors="black"):
     """Make collections from a list of EnergyPlus surfaces."""
-    coords = [
-        getcoords(s) for s in surfaces if s.Surface_Type.lower() == surface_type.lower()
-    ]
+    origin = {}
+    
+    # if coordinate system is relative,
+    # get the zone origin coordinates for each surface
+    if len(zones) >= 1:
+        coords = []
+        for s in surfaces:
```

Comment:
Prefer `for surface in surfaces` outside of a list comp. Single-letter variable names can get confusing in nested for loops. Maybe not confusing here, but it's a style thing I'd rather keep to.

Reply:
Also, please add a test case that covers this new chunk of code.